src/CMakeLists.txt

```diff
@@ -45,53 +40,65 @@ SET(common_src
     SocketBuffer.c
     Heap.c
     LinkedList.c
-    )
+)
 
-IF (CMAKE_SYSTEM_NAME MATCHES "Windows")
-    SET(LIBS_SYSTEM ws2_32)
-ELSEIF (CMAKE_SYSTEM_NAME MATCHES "Linux")
-    SET(LIBS_SYSTEM dl)
-ENDIF()
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+    set(LIBS_SYSTEM ws2_32)
+elseif (CMAKE_SYSTEM_NAME MATCHES "Linux")
+    set(LIBS_SYSTEM dl)
+endif()
 
-ADD_EXECUTABLE(MQTTVersion MQTTVersion.c)
-ADD_LIBRARY(paho-mqtt3c SHARED ${common_src} MQTTClient.c)
-ADD_LIBRARY(paho-mqtt3a SHARED ${common_src} MQTTAsync.c)
-TARGET_LINK_LIBRARIES(paho-mqtt3c pthread ${LIBS_SYSTEM}) 
-TARGET_LINK_LIBRARIES(paho-mqtt3a pthread ${LIBS_SYSTEM})
-TARGET_LINK_LIBRARIES(MQTTVersion paho-mqtt3a paho-mqtt3c ${LIBS_SYSTEM})
-SET_TARGET_PROPERTIES(
+add_executable(MQTTVersion MQTTVersion.c)
+add_library(paho-mqtt3c SHARED ${common_src} MQTTClient.c)
+add_library(paho-mqtt3a SHARED ${common_src} MQTTAsync.c)
+target_include_directories(paho-mqtt3c PUBLIC
+    ${CMAKE_CURRENT_LIST_DIR}
+    ${CMAKE_BINARY_DIR}
+)
+target_include_directories(paho-mqtt3a PUBLIC
+    ${CMAKE_CURRENT_LIST_DIR}
+    ${CMAKE_BINARY_DIR}
+)
+target_link_libraries(paho-mqtt3c pthread ${LIBS_SYSTEM})
+target_link_libraries(paho-mqtt3a pthread ${LIBS_SYSTEM})
+target_link_libraries(MQTTVersion paho-mqtt3a paho-mqtt3c ${LIBS_SYSTEM})
+set_target_properties(
     paho-mqtt3c paho-mqtt3a PROPERTIES
     VERSION ${CLIENT_VERSION}
     SOVERSION ${PAHO_VERSION_MAJOR})
-INSTALL(TARGETS paho-mqtt3c paho-mqtt3a MQTTVersion
+install(TARGETS paho-mqtt3c paho-mqtt3a MQTTVersion
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib)
-INSTALL(FILES MQTTAsync.h MQTTClient.h MQTTClientPersistence.h
-    DESTINATION include)
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+)
+install(FILES MQTTAsync.h MQTTClient.h MQTTClientPersistence.h
+    DESTINATION include
+)
 
-IF (PAHO_WITH_SSL)
-    SET(OPENSSL_LIB_SEARCH_PATH "" CACHE PATH "Directory containing OpenSSL libraries")
```

Comment:
@guilhermeferreira, thanks for having a look, but that does not seem to be the case. See [here](https://travis-ci.org/eclipse/paho.mqtt.c/jobs/166283011), line 47. Travis CI is using plain old GNU make to build paho. I guess that openssl needs to be installed, e.g. by using `brew` or so. Or, maybe the build node has been reinstalled without openssl?